src/ModelContextProtocol.Core/Server/IEventStore.cs

```diff
@@ -0,0 +1,56 @@
+using ModelContextProtocol.Protocol;
+
+namespace ModelContextProtocol.Server;
+
+/// <summary>
+/// Defines the contract for storing and replaying SSE events to support resumability.
+/// </summary>
+/// <remarks>
+/// <para>
+/// When a client reconnects with a <c>Last-Event-ID</c> header, the server uses the event store
+/// to replay events that were sent after the specified event ID. This enables clients to
+/// recover from connection drops without losing messages.
+/// </para>
+/// <para>
+/// Events are scoped to streams, where each stream corresponds to either a specific request ID
+/// (for POST SSE responses) or a special "standalone" stream ID (for unsolicited GET SSE messages).
+/// </para>
+/// <para>
+/// Implementations should be thread-safe, as events may be stored and replayed concurrently.
+/// </para>
+/// </remarks>
+public interface IEventStore
```

Comment:
This is a fairly general name. Could we make it ISseEventStore or something similarly more specific?

Reply:
Good idea. Renamed to `ISseEventStreamStore`.